network/dag/state.go

```diff
@@ -324,36 +325,33 @@ func (s *state) Head(ctx context.Context) (hash.SHA256Hash, error) {
 	return head, err
 }
 
+// Notifier registers receiver under a unique name.
 func (s *state) Notifier(name string, receiver ReceiverFn, options ...NotifierOption) (Notifier, error) {
-	s.notifiersMux.Lock()
-	defer s.notifiersMux.Unlock()
-
-	if _, exists := s.notifiers[name]; exists {
-		return nil, fmt.Errorf("notifier already exists (name=%s)", name)
-	}
 	options = append(options, withCounters(s.eventsNotifyCount, s.eventsFinishedCount))
 
-	notifier := NewNotifier(name, receiver, options...)
-	s.notifiers[name] = notifier
+	n := NewNotifier(name, receiver, options...)
 
-	return notifier, nil
+	_, loaded := s.notifiers.LoadOrStore(name, n)
+	if loaded {
+		return nil, fmt.Errorf("event receiver %q registarion denied on duplicate name", name)
```

Comment:
Why not keep the message?
```suggestion
		return nil, fmt.Errorf("notifier already exists (name=%s)", name)
```

Reply:
* fails to tell which notifier this is about
* fails to tell what the result/decission is
* the (key=value) syntax adds no information
* use quoted strings to detect whitespace and such

Reply:
Ok, but then fix the typo in "registarion" (typos trigger me 🤣 )